src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Triggers/UserTypeArgumentBindingProvider.cs

```diff
@@ -71,32 +72,46 @@ public async Task<ITriggerData> BindAsync(Message value, ValueBindingContext con
                 return new TriggerData(provider, bindingData);
             }
 
-            private static TInput GetBody(Message message, ValueBindingContext context)
+            private static TInput GetBody(Message message)
             {
+                // 1. If ContentType is "application/json" deserialize as JSON.
+                // 2. If ContentType is not "application/json" deserialize the body of a message that was serialized using XmlObjectSerializer.
```

Comment:
How do you know in case 2 that XmlObjectSerializer was used - we don't, right? All we know is it isn't application/json.

Reply:
`message.GetBody<TInput>()` uses XmlObjectSerializer

Reply:
@alrod you're correct that GetBody() works with XmlObjectSerializer serialized body, but the question was how do you know that's the case? Just because it wasn't JSON, doesn't mean it will be XmlObjectSerializer serialized or coming from the legacy ASB client. That's what I don't quite get as well.

Reply:
I will add explanation about `XmlObjectSerializer` (your first comment above).

Reply:
Yeah, just recommend a comment like:

```
// 1. If ContentType is "application/json" deserialize as JSON.
// 2. If ContentType is not "application/json" attempt to deserialize using Message.GetBody, which will handle cases like XML serialization
// 3. If this deserialization fails, do a final attempt attempt at JSON deserialization to catch cases where the content type might be incorrect
```